modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -1,28 +1,200 @@
+            .await
+            .map_err(|e| match e {
+                AutocommitError::CommitFailed {
+                    last_error,
+                    attempts,
+                } => last_error.context(format!("Failed to commit after {attempts} attempts")),
+                AutocommitError::ClosureError { error, .. } => error,
+            })?;
+
+        Ok((operation_id, address))
+    }
+
+    async fn subscribe_deposit_updates(
+        &self,
+        operation_id: OperationId,
+    ) -> anyhow::Result<BoxStream<DepositState>> {
+        let (wallet_client, _) =
+            self.get_first_module::<WalletClientModule>(&WalletCommonGen::KIND);
+
+        let operation_log_entry = self
+            .get_operation(operation_id)
+            .await
+            .ok_or(anyhow!("Operation not found"))?;
+
+        if operation_log_entry.operation_type() != WalletCommonGen::KIND.as_str() {
+            bail!("Operation is not a wallet operation");
+        }
+
+        let operation_meta = operation_log_entry.meta::<WalletOperationMeta>();
+
+        if !matches!(operation_meta, WalletOperationMeta::Deposit { .. }) {
+            bail!("Operation is not a deposit operation");
+        }
+
+        let mut operation_stream = wallet_client.notifier.subscribe(operation_id).await;
+        let tx_subscriber = self.transaction_updates(operation_id).await;
+
+        Ok(Box::pin(stream! {
+            match next_deposit_state(&mut operation_stream).await {
+                Some(DepositStates::Created(_)) => {
+                    yield DepositState::WaitingForTransaction;
+                },
+                Some(DepositStates::TimedOut(_)) => {
+                    yield DepositState::Failed("Deposit timed out".to_string());
+                    return;
+                }
+                Some(s) => {
+                    panic!("Unexpected state {s:?}")
+                },
+                None => return,
+            }
+
+            match next_deposit_state(&mut operation_stream).await {
+                Some(DepositStates::WaitingForConfirmations(_)) => {
+                    yield DepositState::WaitingForConfirmation;
+                },
+                Some(s) => {
+                    panic!("Unexpected state {s:?}")
+                },
+                None => return,
+            }
+
+            let claiming = match next_deposit_state(&mut operation_stream).await {
+                Some(DepositStates::Claiming(claiming)) => claiming,
+                Some(s) => {
+                    panic!("Unexpected state {s:?}")
+                },
+                None => return,
+            };
+            yield DepositState::Confirmed;
+
+            match tx_subscriber
+                .await_tx_accepted(claiming.transaction_id)
+                .await
+            {
+                Ok(_) => {
+                    // FIXME: await claiming ecash and avoid this random delay
+                    sleep(Duration::from_secs(4)).await;
```

Comment:
I think I'll have a fix for this once I submit a PR for this

Reply:
Fixed